src/Compression/CompressionCodecZSTDQAT.cpp

```diff
@@ -0,0 +1,114 @@
+#ifdef ENABLE_ZSTDQAT_COMPRESSION
+#include <CompressionCodecZSTD.h>
+#include <Compression/CompressionInfo.h>
+#include <Compression/CompressionFactory.h>
+#include <zstd.h>
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+#include <Common/typeid_cast.h>
+#include <Poco/Logger.h>
+#include <Common/logger_useful.h>
+#include <qatseqprod.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPRESS;
+    extern const int ILLEGAL_SYNTAX_FOR_CODEC_TYPE;
+    extern const int ILLEGAL_CODEC_PARAMETER;
+}
+
+class CompressionCodecZSTDQAT : public CompressionCodecZSTD
```

Comment:
Since decompression is called much more often in practice than compression, I was wondering if decompression could be hardware-accelerated as well?

Reply:
For now, QAT hardware only support compression. For next generation in future, i will support decompression as well.

Reply:
Done